File: app/src/main/java/xyz/rodit/snapmod/features/opera/SaveMenuOption.java
package xyz.rodit.snapmod.features.opera;

import java.util.Collection;
import java.util.Collections;

import xyz.rodit.snapmod.features.Contextual;
import xyz.rodit.snapmod.features.FeatureContext;
import xyz.rodit.snapmod.mappings.OperaActionMenuOptionViewModel;
import xyz.rodit.snapmod.mappings.OperaContextActions;

public class SaveMenuOption extends Contextual implements MenuPlugin {

    public SaveMenuOption(FeatureContext context) {
        super(context);
    }

    @Override
    public boolean isEnabled() {
        return context.config.getBoolean("allow_download_stories");
    }

    @Override
    public Collection<OperaActionMenuOptionViewModel> createActions() {
        return Collections.singleton(OperaContextActions.getSaveAction());
    }
}

File: app/src/main/java/xyz/rodit/snapmod/features/opera/SaveMenuOption.kt
package xyz.rodit.snapmod.features.opera

import xyz.rodit.snapmod.features.Contextual
import xyz.rodit.snapmod.features.FeatureContext
import xyz.rodit.snapmod.mappings.OperaActionMenuOptionViewModel
import xyz.rodit.snapmod.mappings.OperaContextActions
import xyz.rodit.snapmod.mappings.ParamsMap

class SaveMenuOption(context: FeatureContext) : Contextual(context), MenuPlugin {

    override val isEnabled: Boolean
        get() = context.config.getBoolean("allow_download_stories")

    override fun createActions(params: ParamsMap): Collection<OperaActionMenuOptionViewModel> {
        return setOf(OperaContextActions.getSaveAction())
    }
}